Laravel 5.5 Shift

skibradshaw opened the pull request:
This pull request includes the changes for upgrading to Laravel 5.5. Feel free to commit any additional changes to the `shift-6008` branch.

**Before merging**, you should:

- Checkout the `shift-6008` branch
- Review **all** pull request comments for additional changes
- Update your dependencies for Laravel 5.5
- Run `composer update` (if the pre-scripts fail, add `--no-scripts`)
- Thoroughly test your application

If you need help with your upgrade, check out the [Shift Human Services](https://laravelshift.com/human-services) or join the [Shifty Coders](https://laravelshift.com/shifty-coders) Slack room to get answers to all your Laravel questions.

laravel-shift commented:
Laravel 5.2 modified the default Middleware. Since your middleware differs from the Laravel 5.1 default, you will need to compare the following middleware against the 5.2 versions and merge any changes.

- app/Http/Middleware/Authenticate.php
- app/Http/Middleware/RedirectIfAuthenticated.php

laravel-shift commented:
Laravel 5.2 adds *Middleware Groups* for convenience. Since this may require changes to your routes and controllers, I did not upgrade `app/Http/Kernel.php`. You should review [Middleware Groups](http://laravel.com/docs/5.2/middleware#middleware-groups) in more detail. When ready, merge your `app/Http/Kernel.php` with the [5.2 version](https://github.com/laravel/laravel/blob/becd774e049fb451aca0c7dc4f6d86d7bc12256c/app/Http/Kernel.php) and update your routes/controllers manually.

laravel-shift commented:
Laravel 5.2 condenses the default User model dramatically. Since your `app/User.php` differs from the Laravel 5.1 default, you will need to compare yours against the [5.2 version](https://github.com/laravel/laravel/blob/becd774e049fb451aca0c7dc4f6d86d7bc12256c/app/User.php) and merge any changes.

laravel-shift commented:
Laravel 5.2 updated the `database` session driver. If you are using the `database` driver, you need to add the `user_id` (nullable integer), `ip_address` (nullable string), and `user_agent` (text) columns to your session database table.

laravel-shift commented:
Laravel 5.2 removed the `pretend` mail configuration option. If you were using `pretend`, use the `log` driver instead.

laravel-shift commented:
I see you have custom namespaces in your project. You should consider using the [Consolidate Namespaces Shift](https://laravelshift.com/laravel-consolidate-custom-namespaces) to simplify your namespaces into the default Laravel application namespace.

laravel-shift committed:
Shift Middleware

laravel-shift committed:
Shift core files

laravel-shift committed:
Adopt PSR-2 coding style

The Laravel framework adopts the PSR-2 coding style in version 5.1.
Laravel apps *should* adopt this coding style as well. Read the
[PSR-2 coding style guide][1] for more details and check out [PHPCS][2]
to use as a code formatting tool.

[1]: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
[2]: https://github.com/squizlabs/PHP_CodeSniffer

laravel-shift committed:
Adopt PHP short array syntax

Laravel 5 adopted the short array syntax which became available in
PHP 5.4.

laravel-shift committed:
Shift references to core exceptions

laravel-shift committed:
Shift console commands

laravel-shift committed:
Shift configuration

laravel-shift committed:
Shift Laravel dependencies

skibradshaw commented:
ℹ In Laravel 5.5, the *Request* `has()` method returns `true` for empty strings and `null`.  If you would like to preserve the old behavior of the `has()` method, you may use the new `filled()` method instead.

skibradshaw commented:
ℹ In Laravel 5.5, the *Request* `only()` method returns attributes that are actually present in the request payload. If you would like to preserve the old behavior of the `only()` method, you may use the `all()` method instead.

skibradshaw commented:
❌ Shift could not upgrade the following project files since they differed from the default Laravel version. You will need to compare these project files against the [default Laravel 5.5 versions](https://github.com/laravel-shift/laravel-5.5/) and merge any changes:

- [ ] phpunit.xml
- [ ] tests/Feature/ExampleTest.php

skibradshaw commented:
❌ Shift could not upgrade the following configuration files since they differed from the default Laravel version. You will need to compare these configuration files against the [Laravel 5.5 configuration files](https://github.com/laravel-shift/laravel-5.5/blob/master/config) and merge any changes:

- [ ] config/mail.php

skibradshaw commented:
ℹ Laravel 5.5 now sets the session cookie name using the `APP_NAME`. If you rely on the previous `laravel_session` value, you may set this using the new `SESSION_COOKIE` environment variable. Review the `cookie` key in `config/session.php` for more details.

skibradshaw commented:
❌ Shift detected the *Forms and HTML* package by the Laravel Collective. The recommended version is `5.5.*` for Laravel 5.5. You should review your `laravelcollective/html` dependency to ensure it is compatible. Review the [Installation Docs](https://laravelcollective.com/docs/5.5/html#installation) for more details.

skibradshaw commented:
ℹ Laravel 5.5 changed the default format for HTTP exception responses. This may affect your application if you rely upon default JSON responses. Review the *Exception Format* section of the [Upgrade Guide](https://laravel.com/docs/master/upgrade) for more detail.

skibradshaw commented:
ℹ Laravel 5.5 removed `bootstrap/autoload.php` and uses the Composer `vendor/autoload.php` instead. If you were referencing `bootstrap/autoload.php` directly, you should upgrade your application accordingly.

skibradshaw commented:
ℹ If you are using the `database` cache driver, you should run `php artisan cache:clear` when running your upgraded Laravel 5.5 application for the first time.

skibradshaw commented:
ℹ Laravel 5.5 uses [package auto-discovery](https://medium.com/@taylorotwell/package-auto-discovery-in-laravel-5-5-ea9e3ab20518) to make it easier to install and enable packages.

This may require you to update *Composer*. You may do so by running `composer self-update`. Review the [Composer documentation](https://getcomposer.org/doc/03-cli.md#self-update) for more details.

In addition, you may also unregister the providers and aliases for any packages using auto-discovery from your `config/app.php`.

skibradshaw commented:
ℹ If you have manually upgraded your app from previous versions of Laravel, you should run the [Laravel Linter](https://laravelshift.com/laravel-linter) to ensure your app is fully upgraded to the latest version of Laravel.
